jetty-core/jetty-util/src/main/java/org/eclipse/jetty/util/Fields.java

```diff
@@ -246,13 +248,30 @@ public void add(String name, String... values)
             fields.compute(name, (k, f) ->
             {
                 if (f == null)
-                    return new Field(name, List.of(values));
+                    return new Field(name, toList(values));
                 else
-                    return new Field(f.getName(), f.getValues(), List.of(values));
+                    return new Field(f.getName(), f.getValues(), toList(values));
             });
         }
     }
 
+    static List<String> toList(String... strings)
```

Comment:
This belongs in `StringUtil`

Reply:
Also,  why do the JVM libraries always have `foo()`, `foo(Object o1)`, `foo(Object o1, Object o2)`, ... , `foo(Object o1, Object o2, ... , Object o10)` and `foo(Object... object)` ?        Is there something the JIT can't optimize away with few args calls to a varargs method?

Reply:
@gregw it's because there are well-known shortcomings in the JIT making it unable to eliminate the array allocation. Iterating the array with a variable index, or reading its length are some of those, see https://bugs.openjdk.org/browse/JDK-8332840 for a recent bug on the subject.

Reply:
Moved to `StringUtils`.